### PR TITLE
Speed-related optimizations

### PR DESCRIPTION
**Pull request by dset0x**
Note the change in the structure of requirements due to the addition of the dependency on `backports.functools_lru_cache` for python < 3.2.

The biggest performance improvement comes from the use of `im.getcolors` however.

Overall, processing time has been cut in half.

Before and after times of 23 images, of around 2MP each:
````
41.48s user 0.17s system 99% cpu 41.725 total
19.74s user 0.23s system 99% cpu 20.105 total
```

Before and after times of 64 images, of around 0.3MP each:
```
27.45s user 0.02s system 99% cpu 27.521 total
15.33s user 0.04s system 98% cpu 15.609 total
```

### CODE REVIEW

File: colorific/palette.py [hunk 1]
```diff
@@ -131,8 +139,8 @@ def extract_colors(
     im = autocrop(im, config.WHITE)  # assume white box
     im = im.convert(
         'P', palette=Image.ADAPTIVE, colors=n_quantized).convert('RGB')
-    data = im.getdata()
-    dist = Counter(data)
+    dist = Counter({color: count for count, color
+                    in im.getcolors(n_quantized)})
```
Comment (larsyencken) [inline colorific/palette.py]: Nice, I hadn't noticed that this method was added to images.

**Comment by larsyencken** [comment]
Good work! I'll see that this gets merged, a little admin to do on my side first.

**Comment by dhotson** [comment]
Sorry about the wait. Thanks for the PR! :)

**Comment by dset0x** [comment]
Awesome, thanks for merging this. More to come.

**Pull request merged**